hudi-common/src/test/java/org/apache/hudi/common/testutils/reader/HoodieTestReaderContext.java

```diff
@@ -122,11 +138,9 @@ public HoodieRecord<IndexedRecord> constructHoodieRecord(
     if (!recordOpt.isPresent()) {
       HoodieKey key = new HoodieKey((String) metadataMap.get(INTERNAL_META_RECORD_KEY),
           (String) metadataMap.get(INTERNAL_META_PARTITION_PATH));
-      return new HoodieEmptyRecord<>(
-          key,
-          HoodieOperation.DELETE,
-          (Comparable<?>) metadataMap.get(INTERNAL_META_ORDERING_FIELD),
-          HoodieRecord.HoodieRecordType.AVRO);
+      return new HoodieAvroRecord(key, new EmptyHoodieRecordPayload(
+          null,
+          (Comparable<?>) metadataMap.get(INTERNAL_META_ORDERING_FIELD)));
```

Comment:
After following the logic in `HoodieMergedLogRecordScanner#processNextDeletedRecord`, the delete in the log block leads to a payload of the configured payload class with null record and ordering value, not `EmptyHoodieRecordPayload`.  Let's follow that logic too.

```
SpillableMapUtils
  /**
   * Utility method to convert bytes to HoodieRecord using schema and payload class.
   */
  public static <R> R generateEmptyPayload(String recKey, String partitionPath, Comparable orderingVal, String payloadClazz) {
    HoodieRecord<? extends HoodieRecordPayload> hoodieRecord = new HoodieAvroRecord<>(new HoodieKey(recKey, partitionPath),
        HoodieRecordUtils.loadPayload(payloadClazz, new Object[] {null, orderingVal}, GenericRecord.class, Comparable.class));
    return (R) hoodieRecord;
  }
```